### ISSUE TITLE
Consider passing `detail` as the error message

### ISSUE THREAD

**Issue by richardscarrott** [taxjar/taxjar-node#33]
A lot of Node tooling (e.g. logging services, verror etc.) expect a `message` on an error object however this lib just instantiates an `Error` without a message here: https://github.com/taxjar/taxjar-node/blob/master/lib/util/request.ts#L30

It would be useful if `detail` was the message, e.g. ``new Error(`TaxJar: ${result.error.detail}`)``

Additionally does TaxJar's API return error *code*s as, at least this package appears to just return dynamic error messaging:

```
{
  stack: '...',
  error: 'Bad Request',
  detail: 'to_zip 90210 is not used within to_state CALIFORNIA',
  status: 400
}
```

I'd like to see something like `code: 'invalid_zip'` so we can present localised error messaging on our frontend as it wouldn't be feasible to do so with `detail` because it's dynamic?

**Comment by fastdivision** [comment]
@richardscarrott We'll make this happen. Maybe something similar to this:

```ts
let proxiedError = new (<any>Error)(`TaxJar: ${result.error.error} - ${result.error.detail}`);
```

Our API does not return error specific codes such as `invalid_zip` yet. I'll share this suggestion with our platform team and look into updating our [docs](https://developers.taxjar.com/api/reference/?javascript#errors) with a concrete list of error messages. In the meantime, you could check `detail` for specific keywords in the error message. Thank you for the feedback!

**Comment by richardscarrott** [comment]
@fastdivision it's a shame the API doesn't support error codes so yes it'd be great if you could document the full list of `detail` messages so we can determine the best way to identify an error.

Also, happy to send a PR for the `error.message` if it'd help?

**Comment by ScottRudiger** [comment]
Thanks @richardscarrott, I went ahead and opened a PR to add an error message.

Great suggestions! 👍 We'll keep this open for now so we can update you on the potential addition of specific error codes and docs changes.

**Comment by sethobey** [comment]
Closing issue as it should be resolved with #34
